src/Core/Utils/Attribs.hpp

```diff
@@ -344,18 +432,22 @@ class RA_CORE_API AttribManager
             m_attribs[idx].reset( nullptr );
             m_attribsIndex.erase( c );
         }
-        h.m_idx  = Index::Invalid(); // invalidate whatever!
-        h.m_name = "";               // invalidate whatever!
+        h.m_idx   = Index::Invalid(); // invalidate whatever!
+        auto name = h.m_name;
+        h.m_name  = ""; // invalidate whatever!
+        --m_numAttribs;
+        notify( name );
     }
 
     /// Return true if *this and \p other have the same attributes, same amount
     /// and same names.
     /// \warning There is no check on the attribute type nor data.
     bool hasSameAttribs( const AttribManager& other );
 
-  private:
     /// Perform \p fun on each attribute.
     // This is needed by the user to avoid caring about removed attributes (nullptr)
+    // \todo reimplement as range for
```

Comment:
what about this todo and the next one?

Reply:
It's todo ... as mentionned in the PR comment. It will be done one day, not now.